src/Integration.Vsix/SolutionAnalysisRequester.cs

```diff
@@ -0,0 +1,106 @@
+//-----------------------------------------------------------------------
+// <copyright file="SolutionAnalysisRequester.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Options;
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace SonarLint.VisualStudio.Integration.Vsix
+{
+    internal class SolutionAnalysisRequester
+    {
+        private readonly Workspace workspace;
+
+        private object optionService;
+        private Option<bool> option;
+        private MethodInfo optionServiceSetOptionsMethod;
+        private MethodInfo optionServiceGetOptionsMethod;
+
+        public SolutionAnalysisRequester(Workspace workspace)
+        {
+            if (workspace == null)
+            {
+                throw new ArgumentNullException(nameof(workspace));
+            }
+
+            this.workspace = workspace;
+            this.FindOptionService();
+            this.FindFullSolutionAnalysisOption();
+        }
+
+        private void FindOptionService()
+        {
+            var types = typeof(PerLanguageOption<int>).Assembly.GetTypes();
+            var interfaces = types.Where(t => t.IsInterface);
+
+            var optionServiceType = interfaces.FirstOrDefault(t => t.Name == "IOptionService");
+            if (optionServiceType == null)
+            {
+                return;
+            }
+
+            this.optionService = this.workspace.Services.GetType()
```

Comment:
We probably want asserts to pick up cases when the API has changed. Should we also output to the user that he need to install a newer version of the extension, one will use a proper API/a different hacky API?